test/run-tests

```diff
@@ -377,8 +382,16 @@ def main():
     parser.add_argument("--cache", default="/cache",
                         help="Directory for caching VM images")
     parser.add_argument(
-        "--inventory",
+        "--inventory", help="Inventory to use for VMs",
         default='/usr/share/ansible/inventory/standard-inventory-qcow2')
+    parser.add_argument("pull_request", nargs="*", default=None,
+                        help="Pull requests to test. Example: "
+                        "linux-system-roles/network/1")
+    parser.add_argument(
+        "--use-images", default="*",
+        help="Test pull request only against images matching this pattern")
+    parser.add_argument("--dry-run", help="Do not update pull request status",
+                        default=False, action="store_true")
 
```

Comment:
It seems that with `--dry-run` the code still needs `/secrets` although it maybe should not?

Reply:
Also `--dry-run` apparently also means that the test artifacts are not uploaded.

Reply:
`--dry-run` still needs to use the Github API to get the details of the pull request, therefore I guess it needs a Github token from the secrets. Not sure if it would be possible to get the necessary details without authentication to the API, it might be possible.

Are you suggesting that the artifacts should be uploaded or that it should be mentioned in the help?

Reply:
> --dry-run still needs to use the Github API to get the details of the pull request, therefore I guess it needs a Github token from the secrets. Not sure if it would be possible to get the necessary details without authentication to the API, it might be possible.

It would be great if it were possible, it would make local running of the test harness more easy and useful. Not sure whether it is possible to fetch this information anonymously though.

> Are you suggesting that the artifacts should be uploaded or that it should be mentioned in the help?

The latter.

Reply:
https://developer.github.com/v3/#authentication says "Requests that require authentication" which seems to imply that there are requests which don't. I have not tried it though.